### ISSUE TITLE
`Series.hist` 'category' should probably be `Struct` instead of `Category`?

### ISSUE THREAD

**Issue by MarcoGorelli** [pola-rs/polars#18645]
### Description

Currently with Series.hist we get e.g.

```
shape: (4, 3)
┌────────────┬───────────────┬───────┐
│ breakpoint ┆ category      ┆ count │
│ ---        ┆ ---           ┆ ---   │
│ f64        ┆ cat           ┆ u32   │
╞════════════╪═══════════════╪═══════╡
│ 2.75       ┆ (0.993, 2.75] ┆ 3     │
│ 4.5        ┆ (2.75, 4.5]   ┆ 2     │
│ 6.25       ┆ (4.5, 6.25]   ┆ 0     │
│ 8.0        ┆ (6.25, 8.0]   ┆ 2     │
└────────────┴───────────────┴───────┘
```

`'category'` is a categorical column of strings...wouldn't it be more useful for it to be a `Struct`, with fields `lower_bound` and `upper_bound`, so that the values can be used in further calculations?

**Comment by ritchie46** [comment]
Yes, I think it would. It would also elide the pesky stringcache problem.

**Comment by mcrumiller** [comment]
An issue that will need to be addressed is the interval enclosure. The `category` string indicates whether each bound is open or closed.

This could possibly simply be ignored. It means that if you're handed the output from someone else you cannot determine where values on the endpoint came from, but usually you are the one constructing the result in the first place. Alternatively, the struct could be titled (e.g.) `left_closed` or `left_open`. I do not personally like storing information about the column in the column name itself, but we do not store column metadata aside from the dtype.

Another probably bad idea is to add an `pl.Interval` datatype, which could include a namespace to extract the type. This would allow the resulting interval information to be stored in a single column. Intervals could be useful elsewhere as well. But this would most likely be a lot of work.

Thoughts?

**Comment by Julian-J-S** [comment]
yes please!

would also be nice to really simplify the result.
Instead of `breakpoint(f64)+category(cat)` ONLY return `edges(struct)` (has some thoughts here https://github.com/pola-rs/polars/pull/18761#issuecomment-2370803292)

So imo the result would be:

```
┌──────────────┬───────┐
│ EDGES        ┆ count │
│ ---          ┆ ---   │
│ STRUCT       ┆ u32   │
╞══════════════╪═══════╡
│ {2.0, 3.5}   ┆ 1     │ also notice: 1.994 -> 2 ("nice" edges")
│ {3.5, 5.0}   ┆ 1     │
│ {5.0, 6.5}   ┆ 0     │
│ {6.5, 8.0}   ┆ 1     │
└──────────────┴───────┘
```

This is also what numpy does. Return the correct edges instead of the pandas "hack" that slightly moves the left most edge.

You could then also easily add a `closed` parameter like `rolling` has to specify which side is closed.
